things/service.go

```diff
@@ -172,17 +173,26 @@ func (svc service) ListClients(ctx context.Context, token, reqUserID string, pm
 			if _, err := svc.authorize(ctx, "", auth.UserType, auth.UsersKind, res.GetId(), auth.MembershipPermission, auth.DomainType, res.GetDomainId()); err != nil {
 				return mgclients.ClientsPage{}, err
 			}
-			ids, err = svc.listClientIDs(ctx, res.GetId(), pm.Permission)
-			if err != nil {
-				return mgclients.ClientsPage{}, errors.Wrap(svcerr.ErrNotFound, err)
-			}
+			pm.EntityType = auth.UserType
+			pm.EntityID = res.GetId()
+		}
+
+	default:
+		return mgclients.ClientsPage{}, errors.Wrap(svcerr.ErrMalformedEntity, fmt.Errorf("invalid entity type %s", pm.EntityType))
+	}
+
+	// pm.EntityType will be not empty if entity type is user or channel or if request user is not viewer of domain.
+	if pm.EntityType != "" {
+		thingIDs, err := svc.listThings(ctx, res.GetDomainId(), pm.EntityType, pm.EntityID, pm.Permission)
+		if err != nil {
+			return mgclients.ClientsPage{}, err
 		}
+		pm.IDs = thingIDs
 	}
 
-	if len(ids) == 0 && pm.Domain == "" {
+	if len(pm.IDs) == 0 && pm.Domain == "" {
 		return mgclients.ClientsPage{}, nil
 	}
-	pm.IDs = ids
 	tp, err := svc.clients.SearchClients(ctx, pm)
```

Comment:
@felixgateru , Why we are using `SearchClients` instead of `RetrieveAll`

Reply:
This had been preffered in the latest implementation of list things. But I am checking which will be best for this.